### Hanan Sherka 
### Statistical Analysis of Wages for City of Seattle
### 08.30.2018

## Abstract


In this lab, we took City of Seattle's data on the houlry wages of all its' employees, and found the minimum and maximum hourly wages, and the average. We also looked at who the people with the min and max hourly wages, and the department they worked in/ 

## Dataset Preparation 

I opened the data file of Seattle Wage Data and put it into the empty list called wage data 

In [12]:
#open the data file
wage_file = open("Simplified_City_of_Seattle_Wage_Data.csv", "r")
#Make it an empty list 
wage_data = []
#Going through every line of data and putting it into the list
for line in wage_file:
    wage_data.append(line);
#close file
wage_file.close();

I removed the header columns and split the wage data into two lists, one is the hourly rate and the other is the employee information 

In [13]:
#taking away the column headers 
del wage_data[0]; 

#Make empty lists for the hourly rate and the employee information 
hourly_rate = []
employee_info = [] 

#Going thorugh every line of the list and split the data at the comma 
for row in range(len(wage_data)): 
    column = wage_data[row].split(",")
    
    hourly_rate.append(float(column[0]))
    employee_info.append(column[1])
#Test to veryify that it would print the first four values in the list
print(hourly_rate[:4])
print(employee_info[:4]) 



[53.79, 51.574, 54.84, 25.54]
['Employee: Leslie Kennedy  Job Title: StratAdvsr2 General Govt  Department: Finance & Admin Services Dept\n', 'Employee: Christina Johnson  Job Title: StratAdvsr2 General Govt  Department: Finance & Admin Services Dept\n', 'Employee: Michael Johnson  Job Title: StratAdvsr2 General Govt  Department: Finance & Admin Services Dept\n', 'Employee: Michael Mustello  Job Title: Cust Svc Rep  Department: Finance & Admin Services Dept\n']


## Data Modeling

I used the min and max function to find who makes the least and who makes the most money.  

In [14]:
#Used the min and max function to find the smallest hourly rates and the lowest hourly rates
minwage = min(hourly_rate)
maxwage = max(hourly_rate)

print(minwage)
print(maxwage)

5.53
137.788


In [15]:
#because the min and max function only found the hourly rates and not the information that goes along with that rate
#(like how much that person makes), I had to find the index of the hourly rates so I could find which employee
#that corresponds to 

minindex =  hourly_rate.index(minwage)
maxindex = hourly_rate.index(maxwage)

print(employee_info[minindex])
print(employee_info[maxindex])

Employee: Amy Bonfrisco  Job Title: Civil Svc Commissioner  Department: Civil Service Commissions Dept

Employee: Mami Hara  Job Title: SPU General Mgr&CEO  Department: Seattle Public Utilities



I found the average hourly rate. 

In [16]:
#I divided the sum of all hourly rates by the length of the hourly rate list (aka the number of values in that list) 
#to find the average hourly rate
avgwage = sum(hourly_rate)/len(hourly_rate)
print(avgwage)

39.58501973094151


## Data Analysis and Conclusion 

I found that the City of Seattle employee that makes the most money is Mami Hara, who is the general manager and CEO of Seattle Public Utilities and is paid $137.79 an hour. The employee that makes the least amount of money is Amy Bonfrisco, who works as a Civil Service Comissioner and is paid only $5.53 an hour.I also found that the everage hourly rate is approximately $39.59. This leads me to wonder how someone is paid almost $10 less than Seattle's minimum wage. It also leads me to question what specific factors may influence how much an employee is paid specifically for City of Seattle (ex. how much experience, education, past success in that job, etc.). These are all things up for further exploration in the future.  

## Acknowledgements

I would like to acknowledge Ms. Sconyers for in class support on specific questions I had, and the More-Python-Practice document she made to copy code from, specifically on the minimum and maximum functions, and the dataset preparation. 